cmd/app/run.go

```diff
@@ -22,38 +21,35 @@ func Run() {
 	results := message.Results{Map: make(map[string]bool)}
 	failedCount := 0
 
-	regex, err := regexp.Compile("^(http|https)://")
-	if err != nil {
-		log.Println("Failed to create new regex object")
-		return
-	}
+	socketChan := make(chan socket.Result)
+	var wg sync.WaitGroup
 
-	// iterate over given/loaded sockets
+	// iterate over given/loaded sockets --> start goroutines
 	for _, socket := range list.Sockets {
-		// http/https app protocol patterns check
-		match := regex.MatchString(socket.Host)
-		if !match {
-			// testing raw host and port (tcp), report only unsuccessful tests; exclusively non-HTTP/S sockets
-			rawErr := netrunner.RawConnect(socket)
-			if rawErr != nil {
-				messengerText += fmt.Sprintln(socket.Host, ":", socket.Port, rawErr.Error())
-				failedCount++
-			}
-			results.Map[socket.ID] = (rawErr == nil)
-			continue
-		}
+		wg.Add(1)
+		go netrunner.TestSocket(socket, socketChan, &wg)
```

Comment:
Whenever you're invoking funcs concurrently in a loop make sure to set some sort of limit to make sure you are not spinning up too many goroutines that could be starving app's computational resources leading to a potential crash.